src/ldp/AuthenticatedLdpHandler.ts

```diff
@@ -0,0 +1,145 @@
+ */
+export interface AuthenticatedLdpHandlerArgs {
+  /**
+   * Parses the incoming requests.
+   */
+  requestParser: RequestParser;
+  /**
+   * Extracts the credentials from the incoming request.
+   */
+  credentialsExtractor: CredentialsExtractor;
+  /**
+   * Extracts the required permissions from the generated Operation.
+   */
+  permissionsExtractor: PermissionsExtractor;
+  /**
+   * Verifies if the requested operation is allowed.
+   */
+  authorizer: Authorizer;
+  /**
+   * Executed the operation.
+   */
+  operationHandler: OperationHandler;
+  /**
+   * Writes out the response of the operation.
+   */
+  responseWriter: ResponseWriter;
+}
+
+/**
+ * The central manager that connects all the necessary handlers to go from an incoming request to an executed operation.
+ */
+export class AuthenticatedLdpHandler implements HttpHandler {
+  private readonly requestParser: RequestParser;
+
+  private readonly credentialsExtractor: CredentialsExtractor;
+
+  private readonly permissionsExtractor: PermissionsExtractor;
+
+  private readonly authorizer: Authorizer;
+
+  private readonly operationHandler: OperationHandler;
+
+  private readonly responseWriter: ResponseWriter;
+
+  /**
+   * Creates the handler.
+   * @param args - The handlers required. None of them are optional.
+   */
+  public constructor (args: AuthenticatedLdpHandlerArgs) {
+    Object.assign(this, args);
+  }
+
+  /**
+   * Checks if the incoming request can be handled. The check is very non-restrictive and will usually be true.
+   * It is based on whether the incoming request can be parsed to an operation.
+   * @param input - Incoming request and response. Only the request will be used.
+   *
+   * @returns A promise resolving if this request can be handled, otherwise rejecting with an Error.
+   */
+  public async canHandle (input: { request: HttpRequest; response: HttpResponse }): Promise<void> {
+    return this.requestParser.canHandle(input.request);
+  }
+
+  /**
+   * Handles the incoming request and writes out the response.
+   * This includes the following steps:
+   *  - Parsing the request to an Operation.
+   *  - Extracting credentials from the request.
+   *  - Extracting the required permissions.
+   *  - Validating if this operation is allowed.
+   *  - Executing the operation.
+   *  - Writing out the response.
+   * @param input - The incoming request and response object to write to.
+   *
+   * @returns A promise resolving when the handling is finished.
+   */
+  public async handle (input: { request: HttpRequest; response: HttpResponse }): Promise<void> {
+    let err: Error = null;
+    let operation: Operation = null;
+
+    try {
+      operation = await this.runHandlers(input.request);
+    } catch (error) {
+      err = error;
+    }
+
+    const writeData = { response: input.response, operation, error: err };
+
+    await this.responseWriter.canHandle(writeData);
+
+    return this.responseWriter.handle(writeData);
```

Comment:
Something to keep in mind for the future: if we see this pattern canHandle-handle occuring a lot, we may want to consider simplifying this. But I suspect that more complex cases will occur, where a simplification is not possible.